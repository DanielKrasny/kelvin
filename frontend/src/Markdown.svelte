<script>
import { safeMarkdown } from './markdown.js';
export let content;
</script>

<div class="md">{@html safeMarkdown(content, { breaks: false })}</div>

<style>
:global(.md h1) {
  font-size: 1.5rem;
}
:global(.md h2) {
  font-size: 1.4rem;
}
:global(.md h3) {
  font-size: 1.3rem;
}
:global(.md h4) {
  font-size: 1.2rem;
}
:global(.md h5) {
  font-size: 1.1rem;
}
:global(.md h6) {
  font-size: 1rem;
}
</style>
