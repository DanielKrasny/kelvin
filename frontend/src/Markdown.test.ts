import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/svelte';
import Markdown from './Markdown.svelte';

describe('Markdown', () => {
    it('Should render empty markdown', () => {
        const wrapper = render(Markdown, { content: '' });
        expect(wrapper.container.innerHTML).toMatchSnapshot();
    });
    it('Should render example markdown', () => {
        const wrapper = render(Markdown, {
            content: `# Title 
## Subtitle
### Subsubtitle

Some **bold** text and also *italic* text.
This is _underlined_ and this is ~~strikethrough~~.
Here is some \`inline code\` and here is a code block:
\`\`\`javascript
console.log('Hello, world!');
\`\`\`
Also there is [link](https://example.com) and an image:
![alt text](https://example.com/image.jpg)

1. First
2. Second
3. Third

- One
- Two
- Three

> Blockquote
> is very useful

Also we don't forget about tables:
| Header 1 | Header 2 |
|----------|----------|
| Cell 1   | Cell 2   |
| Cell 3   | Cell 4   |
> which looks like, that we don't currently render :(.
`
        });
        expect(wrapper.container.innerHTML).toMatchSnapshot();
    });
});
