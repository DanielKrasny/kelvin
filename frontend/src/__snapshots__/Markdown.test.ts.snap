// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Markdown > Should render empty markdown 1`] = `"<div class="md"></div>"`;

exports[`Markdown > Should render example markdown 1`] = `
"<div class="md"><h1>Title</h1>
<h2>Subtitle</h2>
<h3>Subsubtitle</h3>
<p>Some <strong>bold</strong> text and also italic text.
This is underlined and this is strikethrough.
Here is some <code>inline code</code> and here is a code block:</p>
<pre><code class="language-javascript">console.log('Hello, world!');
</code></pre>
<p>Also there is <a href="https://example.com">link</a> and an image:
<img src="https://example.com/image.jpg"></p>
<ol>
<li>First</li>
<li>Second</li>
<li>Third</li>
</ol>
<ul>
<li>One</li>
<li>Two</li>
<li>Three</li>
</ul>

<p>Blockquote
is very useful</p>

<p>Also we don't forget about tables:</p>



Cell 1
Cell 2


Cell 3
Cell 4



<p>which looks like, that we don't currently render :(.</p>

</div>"
`;
